test/test_nn.py

```diff
@@ -9637,14 +9637,18 @@ def test_embedding_padding_idx(self, device):
     @dtypesIfCUDA(torch.half, torch.float)
     @dtypes(torch.float)
     def test_softmax_backward(self, device, dtype):
-        sizes = [(0, 10), (32, 20), (10, 0)]
+        # Non-even sizes and non-zero shifts test fallback paths in vectorized kernel
+        sizes = [(0, 10), (32, 20), (10, 0), (31, 20), (32, 21), (31, 23)]
```

Comment:
Wouldn't these small sizes (<1024) be dispatched to specialized kernel?

Reply:
Added new tests